### PR TITLE
Failed event deliveries list

### PR DESCRIPTION
**Pull request by Cloud11PL**
## What type of PR is this?
- [ ] 💅 Refactor
- [ ] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #

## Usage Instructions, Screenshots, Recordings

CleanShot 2024-09-23 at 13 11 37

App install loading state:
CleanShot 2024-09-23 at 11 59 27

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

File: src/apps/components/AppWebhooksDisplay/EventDeliveryItem.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  attempts,
+  lastAttemptDate,
+  hasMore,
+  dataTestId,
+}) => {
+  const intl = useIntl();
+
+  return (
+    <Box key={createdAt} marginBottom={4} data-test-id={dataTestId}>
+      <Box display="grid" __gridTemplateColumns={"1fr 1fr"} paddingX={4}>
```
Comment (poulch) [inline src/apps/components/AppWebhooksDisplay/EventDeliveryItem.tsx:33]:
```suggestion
      <Box display="grid" __gridTemplateColumns="1fr 1fr" paddingX={4}>
```

File: src/apps/components/AppWebhooksDisplay/AppWebhooksAttemptDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          </Text>
+        </Text>
+        <Text display="block" size={3}>
+          <Text color="default2">HTTP</Text>{" "}
+          <Text color="critical1" fontWeight="bold">
+            {responseStatusCode}
+          </Text>
+        </Text>
+        <Text display="block" size={3} marginLeft="auto" fontWeight="bold">
+          {new Date(createdAt).toLocaleString()}
```
Comment (poulch) [inline src/apps/components/AppWebhooksDisplay/AppWebhooksAttemptDetails.tsx:47]: queston: why not use component for this like that `<DateTime plain date={createdAt} />`

File: src/apps/components/AppWebhooksDisplay/EventDeliveryItem.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      </Box>
+      {attempts.length > 0 && (
+        <Box marginBottom={2} paddingX={4}>
+          <Text>
+            {intl.formatMessage({
+              defaultMessage: "Attempts:",
+              id: "OFTsI1",
+            })}{" "}
+            <Text size={4} fontWeight="bold">
+              {attemptsCount} / 6
```
Comment (poulch) [inline src/apps/components/AppWebhooksDisplay/EventDeliveryItem.tsx:49]: suggestion: `6` should be constant `MAX_ATTEMPTS_COUNT`

**Review by poulch — APPROVED** [review]

File: src/apps/components/AppWebhooksDisplay/EventDeliveriesList.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+          status={ed.node.status}
+          attemptsCount={attemptsCount}
+          attempts={attempts}
+          lastAttemptDate={lastAttemptDate}
+          hasMore={hasMore}
+        />
+      );
+    })}
+  </>
+);
```
Comment (andrzejewsky) [inline src/apps/components/AppWebhooksDisplay/EventDeliveriesList.tsx:39]: In cases like this you could create something like DeliveryItem class or (even function + interfcace) and EventDeliveryItem component could take a single argument of this type.

That class can have static method eg. `createFrom(deliveryEntry: DeliveryEntry)` - in that case you have elegant separation and you are able to simply call:

```ts
{eventDeliveries.map(entry => <EventDeliveryItem  deliveryItem={DeliveryItem.from(entry)}  />)}
```

and this component is not needed anymore.

Think about that in the future, not needed now.

File: src/apps/components/AppWebhooksDisplay/EventDeliveryItem.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  createdAt: string;
+  attemptsCount: number;
+  status: EventDeliveryStatusEnum;
+  attempts: EventDeliveryAttemptFragment[];
+  lastAttemptDate: string;
+  hasMore: boolean;
+  dataTestId?: string;
+}
+
+const MAX_ATTEMPTS = 6;
```
Comment (andrzejewsky) [inline src/apps/components/AppWebhooksDisplay/EventDeliveryItem.tsx:20]: would be nice to link some docs here so we know where this value come from

**Review by andrzejewsky — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**